### PR TITLE
Towards standard BlobEntry

### PR DESCRIPTION
**Pull request by dehann**
Changes from long engineering meeting today.

fixing 
- #954

### CODE REVIEW

File: src/DataBlobs/entities/AbstractDataEntries.jl [hunk 1]
```diff
@@ -22,6 +22,8 @@ General Data Store Entry.
 """
 @Base.kwdef struct BlobStoreEntry <: AbstractDataEntry
     id::Union{UUID, Nothing}=nothing
+    blobId::Union{UUID, Nothing}=nothing
+    originId::UUID
     label::Symbol
     blobstore::Symbol
     hash::String # Probably https://docs.julialang.org/en/v1/stdlib/SHA
```
Comment (GearsAD) [inline src/DataBlobs/entities/AbstractDataEntries.jl:29]: Make metadata:
```
metadata::Dict{Symbol, SmallDataTypes}
```

File: src/DataBlobs/entities/AbstractDataEntries.jl [hunk 2]
```diff
@@ -22,6 +22,8 @@ General Data Store Entry.
 """
 @Base.kwdef struct BlobStoreEntry <: AbstractDataEntry
```
Comment (GearsAD) [inline src/DataBlobs/entities/AbstractDataEntries.jl:23]: BlobEntry

Comment (dehann) [reply src/DataBlobs/entities/AbstractDataEntries.jl:23]: xref:
- #955

File: src/DataBlobs/entities/AbstractDataEntries.jl [hunk 3]
```diff
@@ -22,6 +22,8 @@ General Data Store Entry.
...
     id::Union{UUID, Nothing}=nothing
+    blobId::Union{UUID, Nothing}=nothing
```
Comment (dehann) [inline src/DataBlobs/entities/AbstractDataEntries.jl:25]:
```suggestion
```

File: src/DataBlobs/entities/AbstractDataEntries.jl [hunk 4]
```diff
@@ -22,6 +22,8 @@ General Data Store Entry.
...
+    originId::UUID
```
Comment (dehann) [inline src/DataBlobs/entities/AbstractDataEntries.jl:26]:
```suggestion
```

File: src/DataBlobs/entities/AbstractDataEntries.jl [hunk 5]
```diff
@@ -20,20 +20,28 @@ export BlobStoreEntry
     $(TYPEDEF)
 General Data Store Entry.
 """
-@Base.kwdef struct BlobStoreEntry <: AbstractDataEntry
-    id::Union{UUID, Nothing}=nothing
+@Base.kwdef struct BlobEntry <: AbstractDataEntry
+    """ This is created by server-side GraphQL """
+    id::Union{UUID, Nothing}=nothing 
+    """ This is the S3 blob ID, or the filesystem blob ID. """
```
Comment (dehann) [inline src/DataBlobs/entities/AbstractDataEntries.jl:26]:
```suggestion
    """ This is the forced server generated blobId, or the filesystem blobId. """
```

File: src/DataBlobs/entities/AbstractDataEntries.jl [hunk 6]
```diff
@@ -14,115 +14,98 @@ getTimestamp(entry::AbstractDataEntry) = entry.timestamp
...
+    """ This is the ID at creation at the edge, do whatever you want with this, but make sure you populate it. """
+    originId::UUID
     label::Symbol
     blobstore::Symbol
     hash::String # Probably https://docs.julialang.org/en/v1/stdlib/SHA
     origin::String # E.g. user|robot|session|varlabel
     description::String
     mimeType::String
-    metadata::String = ""
+    metadata::Dict{Symbol, SmallDataTypes} = Dict{Symbol,SmallDataTypes}()
```
Comment (dehann) [inline src/DataBlobs/entities/AbstractDataEntries.jl:36]:
```suggestion
    metadata::String = ""
```

**Pull request merged**